Code-the-Dream-School/dd-prac-team4-front
# PS-85 Create a checkout page

**Checkout page created**

**Pull Request Description:**

This pull request contains next new following files: 
src/components/Purchase/CheckoutForm.js
src/components/Purchase/CheckoutForm.module.css
src/components/Purchase/CheckoutPage.js
src/components/Purchase/CheckoutPage.module.css

src/components/Purchase/CheckoutPage.js: This file contains a React component that represents the checkout process page. It retrieves the client secret for the payment intent from the server based on the payment amount, and then renders the CheckoutForm component with the appropriate options.

src/components/Purchase/CheckoutForm.js: This file contains a React component that represents the checkout form for making a payment. It handles the Stripe payment integration and displays the payment element and relevant messages based on the payment status.

**Please ensure that the necessary environment variables(like Stripe secret key) and dependencies are properly set up before testing the pull request.**

**Testing Steps:**

- Install the required dependencies using npm install.
Install:

```
npm install 
npm install @stripe/react-stripe-js @stripe/stripe-js
```

- **Make sure the Stripe API keys are properly configured in the environment variables.**
- Run the application locally using npm start.
- Open your browser and access the checkout page.

[http://localhost:3000/checkout](http://localhost:3000/checkout)

- Submit the payment form and verify that the validation messages are displayed correctly for invalid inputs.
- If the input is valid, the checkout process page should be loaded.
- Verify that the checkout form is displayed and functional.
- Test the Stripe integration by entering valid **testing payment details** and completing the payment.

> **Testing payment details:**
> Email: any
> Card number: 
> 4242 4242 4242 4242
> Expiration: any(but date should be in future)
> CVC: any
> Country: United States
> Zip: any

- Test different scenarios, such as successful payment, processing payment, and failed payment, to ensure proper handling of different payment statuses.
- Test the overall flow of the checkout process from start to finish, ensuring that the data is accurately processed and the user is redirected appropriately.

<img width="1426" alt="Screenshot 2023-08-23 at 9 17 42 PM" src="https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/2952900/ac851169-2048-4dd3-8c35-1b923fd974f7">
